app/models/paste.rb

```diff
@@ -99,10 +106,19 @@ def train_classifier
     return if marked_by.nil?
 
     classifier = Rails.application.config.classifier
-    classifier.train marked_kind, content.attachment.open(&:read).force_encoding('utf-8')
+    classifier.train marked_kind, code
   end
 
   def delete_spam
     destroy! if marked_by.present? && marked_kind == 'spam'
   end
+
+  def check_terms
+    Term.all.each do |term|
+      content = self.send(term.subject)
+      next unless content.include?(term.content)
+
+      self.marked_kind = 'spam'
```

Comment:
Just wondering - will this a) deny the paste or b) allow it, and "just" flag it as spam?

I'd very much prefer a) because it will save us lots of manual work.

Reply:
It just marks it as spam, there's nothing stopping me from having both behaviours per term though

Reply:
The thought was mostly to remove it from being publicly listed on the recent pastes for anyone but admins and moderators, but that's fair